packages/mui-joy/src/AccordionDetails/AccordionDetails.test.tsx

```diff
@@ -85,5 +85,47 @@ describe('<AccordionDetails />', () => {
       expect(screen.getByRole('button')).to.have.attribute('aria-expanded', 'true');
       expect(screen.getByTestId('textbox')).to.have.property('tabIndex', 2);
     });
+
+    it('should retain the default tab index if not explicitly set', () => {
+      render(
+        <Accordion>
+          <AccordionSummary>title</AccordionSummary>
+          <AccordionDetails>
+            <input data-testid="textbox" />
+          </AccordionDetails>
+        </Accordion>,
+      );
+
+      expect(screen.getByTestId('textbox')).to.have.property('tabIndex', -1);
+
+      fireEvent.click(screen.getByRole('button')); // open
+
+      expect(screen.getByTestId('textbox')).to.have.property('tabIndex', 0);
+
+      fireEvent.click(screen.getByRole('button')); // close
+
+      expect(screen.getByTestId('textbox')).to.have.property('tabIndex', -1);
+
+      fireEvent.click(screen.getByRole('button')); // reopen
+
+      expect(screen.getByTestId('textbox')).to.have.property('tabIndex', 0);
+    });
+
+    it('should retain the -1 tab index when explicitly set', () => {
+      render(
+        <Accordion>
+          <AccordionSummary>title</AccordionSummary>
+          <AccordionDetails>
+            <input data-testid="textbox" tabIndex={-1} />
+          </AccordionDetails>
+        </Accordion>,
+      );
+
+      expect(screen.getByTestId('textbox')).to.have.property('tabIndex', -1);
+
+      fireEvent.click(screen.getByRole('button')); // open
+
+      expect(screen.getByTestId('textbox')).to.have.property('tabIndex', -1);
```

Comment:
For completeness, I think we should also check that it keeps being `-1` when closed.

```suggestion
      expect(screen.getByTestId('textbox')).to.have.property('tabIndex', -1);
      
      fireEvent.click(screen.getByRole('button')); // close

      expect(screen.getByTestId('textbox')).to.have.property('tabIndex', -1);
```